packages/admin/admin/src/dataGrid/DataGridColumnsManagement.tsx

```diff
@@ -0,0 +1,92 @@
+// MUI's implementation: https://github.com/mui/mui-x/blob/v7.x/packages/x-data-grid/src/components/columnsManagement/GridColumnsManagement.tsx
```

Comment:
It seems to me that MUI's implementation is more sophisticated than ours. Did you intentionally reduce the implementation? Would it be possible to copy MUI's implementation and only swap the checkboxes with switches?